pick up $MPICC and $MPICXX set by EasyBuild, add -DMPICH_IGNORE_CXX_SEEK and -DMPICH_SKIP_MPICXX compiler flags,
add special case of EASYBUILDMPI
author: Kenneth Hoste (HPC-UGent)

diff -ruN foam-extend-4.1.orig/wmake/rules/linux64Icc/mplibEASYBUILDMPI foam-extend-3.0/wmake/rules/linux64Icc/mplibEASYBUILDMPI
--- foam-extend-4.1.orig/wmake/rules/linux64Icc/mplibEASYBUILDMPI	1970-01-01 01:00:00.000000000 +0100
+++ foam-extend-4.1/wmake/rules/linux64Icc/mplibEASYBUILDMPI	2014-02-27 11:48:54.290303826 +0100
@@ -0,0 +1,3 @@
+PFLAGS     = -DMPICH_IGNORE_CXX_SEEK
+PINC       = -DMPICH_SKIP_MPICXX
+PLIBS      = 
diff -ruN foam-extend-4.1.orig/wmake/rules/linux64Gcc/mplibEASYBUILDMPI foam-extend-3.0/wmake/rules/linux64Icc/mplibEASYBUILDMPI
--- foam-extend-4.1.orig/wmake/rules/linux64Gcc/mplibEASYBUILDMPI	1970-01-01 01:00:00.000000000 +0100
+++ foam-extend-4.1/wmake/rules/linux64Gcc/mplibEASYBUILDMPI	2014-02-27 11:48:54.290303826 +0100
@@ -0,0 +1,3 @@
+PFLAGS     = -DMPICH_IGNORE_CXX_SEEK
+PINC       = -DMPICH_SKIP_MPICXX
+PLIBS      = 
--- foam-extend-4.1.orig/wmake/rules/linux64Gcc/c.orig	2014-09-04 14:14:31.446294000 +0200
+++ foam-extend-4.1/wmake/rules/linux64Gcc/c	2014-09-04 14:15:19.846263000 +0200
@@ -2,7 +2,7 @@
 
 cWARN        = -Wall
 
-cc          = gcc -m64
+cc          = $(MPICC)
 
 include $(RULES)/c$(WM_COMPILE_OPTION)
 
--- foam-extend-4.1.orig/wmake/rules/linux64Gcc/c++.orig	2020-02-21 11:41:49.266435000 +0100
+++ foam-extend-4.1/wmake/rules/linux64Gcc/c++	2020-02-21 11:45:30.525280000 +0100
@@ -5,7 +5,7 @@
 # Suppress some warnings for flex++ and CGAL
 c++LESSWARN = -Wno-old-style-cast -Wno-unused-local-typedefs -Wno-array-bounds
 
-CC          = g++ -std=c++11 -m64
+CC          = $(MPICXX) -std=c++11
 
 include $(RULES)/c++$(WM_COMPILE_OPTION)
 
--- foam-extend-4.1.orig/etc/settings.sh.orig	2020-02-18 13:30:37.000000000 +0100
+++ foam-extend-4.1/etc/settings.sh	2020-02-21 11:41:49.268762507 +0100
@@ -411,6 +411,11 @@
     unset mpi_version
     ;;
 
+EASYBUILDMPI)
+    export FOAM_MPI=mpi
+    export FOAM_MPI_LIBBIN=$FOAM_LIBBIN/mpi
+    ;;
+
 MPICH)
     mpi_version=mpich-1.2.4
     export MPI_HOME=$WM_THIRD_PARTY_DIR/$mpi_version
